### DISCUSSION TITLE
Should I add LinearVelocity to a RigidBody?

### DISCUSSION THREAD

**Discussion by rcorre** [avianphysics/avian#325] [Q&A]
[The docs](https://docs.rs/bevy_xpbd_3d/latest/bevy_xpbd_3d/components/enum.RigidBody.html#creation) say:

> Creating a rigid body is as simple as adding the [RigidBody](https://docs.rs/bevy_xpbd_3d/latest/bevy_xpbd_3d/components/enum.RigidBody.html) component:
> Bevy XPBD will automatically add any missing components, like the following:
> ...
> [LinearVelocity](https://docs.rs/bevy_xpbd_3d/latest/bevy_xpbd_3d/components/struct.LinearVelocity.html)

However, the following code will panic with `NoEntities`:

```
use bevy::prelude::*;
use bevy_xpbd_3d::prelude::*;

#[derive(Component)]
struct Player;

#[derive(Resource)]
struct Animations(Vec<Handle<AnimationClip>>);

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_plugins(PhysicsPlugins::default())
        .add_systems(Startup, setup)
        .add_systems(Update, bevy::window::close_on_esc)
        .add_systems(Update, keyboard_movement)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn((
        Player,
        SpatialBundle {
            transform: Transform::from_xyz(4.0, 0.0, 4.0),
            ..Default::default()
        },
        RigidBody::Kinematic,
        // LinearVelocity::default(),
    ));
}

fn keyboard_movement(mut vel: Query<&mut LinearVelocity, With<Player>>) {
    let _ = vel.single_mut();
}
```

If I uncomment the line that explicitly adds a LinearVelocity, it works. Am I misreading the docs?

**Comment by Jondolf** [comment]
The crate does add `LinearVelocity`, but this currently happens at the start of the physics schedule, which by default runs in `PostUpdate`. So there's a slight delay, which in this case causes a crash when expecting the component to exist in `Update` already.
